### PR TITLE
Updated @vue-storefront/nuxt to latest sdk version & set cdnCacheBustingId as optional for sdk

### PR DESCRIPTION
**Pull request by michaelKurowski**
…ingId option in sdk as optional

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSDoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .changeset/purple-kangaroos-shake.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@vue-storefront/nuxt": major
+"@vue-storefront/sdk": minor
+---
+
+cdnCacheBustingId option of sdk is now optional, updated nuxt to the latest sdk version
```
Comment (grixu) [inline .changeset/purple-kangaroos-shake.md:6]:
```suggestion
[CHANGED]: cdnCacheBustingId option of sdk is now optional
[CHANGED]: updated nuxt to the latest sdk version
```

File: packages/nuxt/src/module.ts [hunk 2]
```diff
@@ -67,6 +67,18 @@ export type SdkConfig = ${genInlineTypeImport(
       from: localResolver.resolve("./runtime/utils/composeMiddlewareUrl"),
     });
 
+    addImports({
+      name: "composeMiddlewareUrl",
+      as: "composeMiddlewareUrl",
+      from: localResolver.resolve("./runtime/utils/composeMiddlewareUrl"),
+    });
```
Comment (grixu) [inline packages/nuxt/src/module.ts:74]: Why is this duplicated (lines 64-68 are identical)?

File: packages/nuxt/tsconfig.json [hunk 3]
```diff
@@ -1,3 +1,8 @@
 {
-  "extends": "./.nuxt/tsconfig.json"
+  "extends": "./.nuxt/tsconfig.json",
+  "compilerOptions": {
+    "paths": {
+      "@storefront/shared": ["../shared"]
+    }
+  },
```
Comment (grixu) [inline packages/nuxt/tsconfig.json:7]: `@storefront/shared` is package & devDependency - I believe that change isn't needed

File: packages/nuxt/src/runtime/defineSdkConfig.template [hunk 4]
```diff
@@ -12,6 +12,7 @@ type InjectedContext = {
    */
   getCookieHeader: () => Record<string, string>;
   getRequestHeaders: () => Record<string, string>;
+  defaults: ReturnType<typeof getDefaults>
```
Comment (grixu) [inline packages/nuxt/src/runtime/defineSdkConfig.template:15]: As this is an another quite static configuration data - make this as config object:

```suggestion
  config: {
    middlewareUrl: string;
    defaults: ReturnType<typeof getDefaults>
  }
```

And mark original one `middlewareUrl: string;` as deprecated

Comment (grixu) [reply packages/nuxt/src/runtime/defineSdkConfig.template:15]: Anyway to speed this up right now, we can skip deprecation, but only put it into the object ;)

File: .changeset/purple-kangaroos-shake.md (hunk 1 above)
Comment (michaelKurowski) [reply .changeset/purple-kangaroos-shake.md]: Sorry I've forgot to push a fix for that

File: packages/nuxt/tsconfig.json (hunk 3 above)
Comment (michaelKurowski) [reply packages/nuxt/tsconfig.json:7]: Didn't work otherwise :(

File: packages/sdk/src/modules/middlewareModule/utils/getRequestSender.ts [hunk 5]
```diff
@@ -22,7 +22,7 @@ export const getRequestSender = (options: Options): RequestSender => {
     ssrApiUrl,
     defaultRequestConfig = {},
     methodsRequestConfig = {},
-    cdnCacheBustingId,
+    cdnCacheBustingId = null,
```
Comment (grixu) [inline packages/sdk/src/modules/middlewareModule/utils/getRequestSender.ts:25]: According to the types - shouldn't it be `undefined`?

```suggestion
    cdnCacheBustingId = undefined,
```

**Review by grixu — APPROVED** [review]

**Pull request merged**